tower-reconnect/src/lib.rs

```diff
@@ -52,26 +56,27 @@ impl<M, Target, S, Request> Service<Request> for Reconnect<M, Target>
 where
     M: Service<Target, Response = S>,
     S: Service<Request>,
+    M::Future: Unpin,
```

Comment:
This bound is pretty sad, but needs to be there since `poll_ready` needs to poll that future, and only has `&mut self`. We _can_ `Box::pin` that future if the bound makes us sad. See also #319.

Reply:
So I think in this case we can provide a `MapBoxService` that boxes all the outgoing futures and can take some Servce that produces a !Unpin future and pin it. So that this choice can be done by the User. So in this case I'm 👍 on Unpin here.